benchmarks/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+
+##########################################################
+# Google benchmark
+
+ExternalProject_Add(
+    google_benchmark
```

Comment:
I would agree on not needing another dep. Folly is already a big dep. However, the use of GB is conditional on the benchmarks being built and will only be brought in when needed. And is self contained in the build via the snapshot.

- Cosmetically, GB and Folly are close. But the setup/teardown are not the same. Googles' is much more mature, IMO.
- Measurement technique is similar. However, GB is a closer fit to getting a more relevant benchmark with its sampling technique and run to min time approach.
- Custom counters are relied upon. Not seeing how those are integrated with folly.
- harness should be self contained and not rely on external components. Especially ones that are in use by the unit under test.